### PR TITLE
[MU4] Palette step 4 -  moved palette qml to new ui engine 

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: framework/ui/view/qmltooltip.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        m_item = item;
+        m_text = text;
+
+        if (m_item) {
+            connect(m_item, &QObject::destroyed, this, &QmlToolTip::doHide);
+
+            const int interval = item ? qApp->styleHints()->mousePressAndHoldInterval() : 100;
+            m_timer.start(interval);
+        } else {
+            doHide();
```
Comment (vpereverzev) [inline framework/ui/view/qmltooltip.cpp:55]: It's not really obvious that a method called a "show" can also "hide" something
Maybe "toggle" would be a more convenient name for the method

File: framework/uicomponents/view/iconview.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#ifndef MU_FRAMEWORK_ICONVIEW_H
+#define MU_FRAMEWORK_ICONVIEW_H
+
+#include <QQuickPaintedItem>
+#include <QIcon>
+#include <QColor>
+
+namespace mu {
+namespace framework {
+class IconView : public QQuickPaintedItem
```
Comment (vpereverzev) [inline framework/uicomponents/view/iconview.h:29]: Why did you need another component for the icons?

Comment (igorkorsukov) [reply framework/uicomponents/view/iconview.h:29]: Paletеу symbols are return to Qml as QIcon, on which they are drawn. that is, this component can show the QIcon object passed by value, and not by URL or as code from a font. Perhaps it is worth making this component only for palettes, and not general, and call it something else?

File: framework/ui/view/qmltooltip.cpp (hunk 1 above)
Comment (igorkorsukov) [reply framework/ui/view/qmltooltip.cpp:55]: what he is hiding is more likely an exception handling rather than normal behaviour.
